bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
What is the way to distinguish between "tight" bounds policy and "nice fit" bounds policy?  (Recall that "nice fit" is actually contingent on screen space transforms, because what's nice for a linear scale is not the same as what's nice for a log scale.)